etcdctl/ctlv3/command/check.go

```diff
@@ -265,6 +270,30 @@ func newCheckPerfCommand(cmd *cobra.Command, args []string) {
 	}
 }
 
+func attemptCleanup(client *v3.Client) {
+	dctx, dcancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer dcancel()
+	_, err := client.Delete(dctx, checkPerfPrefix, v3.WithPrefix())
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+}
+
+func interruptableContext(ctx context.Context, attemptCleanup func()) context.Context {
+	ctx, cancel := context.WithCancel(ctx)
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	go func() {
+		defer signal.Stop(signalChan)
+		select {
+		case <-signalChan:
+			cancel()
+			attemptCleanup()
+		}
+	}()
+	return ctx
```

Comment:
Please return cancel here, as:

https://golang.org/pkg/context/#WithCancel

states: 
> Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete.

so we need to call cancel even if there was no 'signal.